### DISCUSSION TITLE
Esecuzione comando NUT in docker - access denied

### DISCUSSION THREAD

**Discussion by SkunkGuru** [indomus/forum#2921] [GENERALE - Discussioni generiche]
Ciao a tutti

ho seguito la guida
https://indomus.it/guide/gestire-un-ups-con-mini-pc-intel-nuc-computer-tramite-nut-network-ups-tools-con-docker-su-linux-debian/#comandi
e funziona tutto correttamente da qualche mese.

Oggi ho deciso di disattivare il beeper dell'ups in caso di assenza di rete, dal momento che mi arriva la notifica da HA.
Ho quindi seguito la parte di guida per l'invio dei comandi nut, ma restituisce sempre errore

ho per prima cosa verificato che il comando fosse supportato
![image](https://github.com/indomus/forum/assets/15073199/26ec141d-61cf-4c3a-972f-30fe9d3dfb37)

l'invio del comando però dà errore access denied
![image](https://github.com/indomus/forum/assets/15073199/1a0875ec-a25f-4315-8eba-7ece3e37abe1)

la password ovviamente è quella corretta, inserita nel file docker-compose, del resto l'integrazione HA funziona (e anche il NAS ci si collega); ho provato sia admin, che root, che master... 

qualche idea ? sto sbagliando qualcosa ? online non riesco a trovare granchè

**Comment by snakuzzo** [comment]
https://nut-upsuser.alioth.debian.narkive.com/QItp695e/err-access-denied-in-upsrw

http://www.networkupstools.org/docs/man/upsd.users.html

**Reply by SkunkGuru** [reply]
ciao,
avevo già provato a seguire quella strada, ma stesso esito

![image](https://github.com/indomus/forum/assets/15073199/7c8c140c-8a0d-44f7-b521-2ca87484789a)

**Reply by SkunkGuru** [reply]
ciao,
ma solo per capire, mi hai indicato quei link perchè sai che con quel metodo si risolve (e in tal caso continuerò a provare quella strada), oppure mi hai copiato 2 link a caso da google ?

**Reply by KC-inDomus** [reply]
la seconda, quasi sicuramente :) se ti ha indicato la strada quella è una possibile soluzione che tu devi verificare

**Reply by SkunkGuru** [reply]
si, come scritto l'ho già verificata ma senza successo al momento

ma davvero nessuno ha mai voluto disattivare l'avviso acustico ?

**Reply by snakuzzo** [reply]
Allora, ho verificato in prima persona con NUT ed il mio UPS APC.
Ecco come fare a disabilitare il beeper (testato e funzionante)...

```
docker exec -it nut sh
```

Controllare la lista di comandi disponibili per il tuo UPS (nel mio caso è un apc)

```
upscmd -l apc
```

e per il beeper mi risultano questi

```
beeper.disable - Disable the UPS beeper
beeper.enable - Enable the UPS beeper
beeper.mute - Temporarily mute the UPS beeper
beeper.off - Obsolete (use beeper.disable or beeper.mute)
beeper.on - Obsolete (use beeper.enable)
```

A questo punto devi editare il tuo file `upsd.users` (visto che sei all'interno del container e non hai `nano`, puoi usare `vi`)...

```
vi /etc/nut/upsd.users
```

Ora, visto che di base avrai definito un solo utente con i permessi per il monitoraggio (perché così è fatta l'immagine docker che stai usando), come specificato sulla [documentazione ufficale di NUT](https://networkupstools.org/docs/man/upsd.users.html), devi aggiungere un utente che abbia i permessi per modificare i parametri, altrimenti avrai sempre `ACCESS DENIED`

Quindi in fondo aggiungi questo utente...

```
[admin]
        password = admin
        actions = set
        actions = fsd
        instcmds = all
```

Esci da `vi` salvando premendo `ESC`, poi in sequenza `:wq!` e poi `INVIO`

Ricarichi il servizio, che prenderà la nuova configurazione

```
upsd -c reload
```

A questo punto...

```
upscmd -u admin -p admin apc beeper.disable
```

se tutto sarà andato bene, ti darà in output

```
OK
```

Per verificare lo stato del beeper...

```
upsc apc ups.beeper.status
```

che ti restituirà

```
disabled
```

e lo vedrai anche sull'integrazione NUT su HA

<img width="372" alt="image" src="https://github.com/indomus/forum/assets/7205371/29dc3a86-9847-424c-91e9-209430b21ca4">

Ovviamente, avendo modificato un file che non è persistente con il bind di volumi e/o filesystem, nel momento in cui ricreerai il container (perché ci sarà un'aggiornamento dell'immagine ad esempio), perderai la modifica appena fatta. Quindi se vuoi, dovrai renderlo persistente con il bind quando crei il container (a meno che tu non riesca a trovare un'immagine docker diversa che preveda anche un utente admin)

**Reply by SkunkGuru** [reply]
grazie mille ! ha funzionato perfettamente 👍 
in realtà è esattamente quello che avevo già provato io, tranne il comando   upsd -c reload
ecco perchè il nuovo utente che avevo aggiunto non stava funzionando
in quel link della documentazione non era indicato  😞

**Reply by snakuzzo** [reply]
Allora, visto che avevi fatto tutto, ti bastava riavviare il container docker :)

**Reply by SkunkGuru** [reply]
in realtà l'avevo pensato che servisse un qualche riavvio per rileggere il nuovo utente, ma temo d'aver usato il comando sbagliato; avevo rilanciato il docker compose up, ma mi sa che dovevo usare il docker compose restart 😭 

di nuovo grazie mille per il tuo tempo 👍

**Reply by snakuzzo** [reply]
```
docker compose restart nut
```

Figurati...ho visto una cosa nuova anche io, non mi ero mai posto il problema prima d'ora ;)
